### PR TITLE
Add documentation about ssl cert per listener

### PR DESCRIPTION
**Pull request by bartam1**
| Q               | A
| --------------- | ---
| New feature?    | yes

### What's in this PR?
Documentation about how to add own generated SSL certificate per listener

### CODE REVIEW

File: docs/ssl.md [hunk 1]
```diff
@@ -23,10 +26,41 @@ If `sslSecrets.create` is `false`, the operator will look for the secret at `ssl
...
 
+## Using own certificates
+### For listener which is not used for inner broker communication
+In this **KafkaCluster** custom resource (https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster_with_ssl_hybrid_customcert.yaml) SSL is enabled for all listeners and certificates will be auto generated for "inner" and "controller" listeners. The "external" and "internal" listeners will use the user provided certificates. **ServerSSLCertSecret** is a reference to the kubernetes secret that contains the server certificate for the listener to be used for SSL communication.
+In the server secret these keys - values must be set:
+
+| Key              | Value                                     |
+|:----------------:|:------------------------------------------|
+| `keystore.jks`   | Certificate and private key in JKS format |
+| `truststore.jks` | Truested CA certificate in JKS format     |
```
Comment (baluchicken) [inline docs/ssl.md:37]:
```suggestion
| `truststore.jks` | Trusted CA certificate in JKS format     |
```

File: docs/ssl.md [hunk 2]
```diff
@@ -23,10 +26,41 @@ If `sslSecrets.create` is `false`, the operator will look for the secret at `ssl
 | `caCert`     | The CA certificate |
 | `caKey`      | The CA private key |
 
+## Using own certificates
+### For listener which is not used for inner broker communication
+In this **KafkaCluster** custom resource (https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster_with_ssl_hybrid_customcert.yaml) SSL is enabled for all listeners and certificates will be auto generated for "inner" and "controller" listeners. The "external" and "internal" listeners will use the user provided certificates. **ServerSSLCertSecret** is a reference to the kubernetes secret that contains the server certificate for the listener to be used for SSL communication.
```
Comment (baluchicken) [inline docs/ssl.md:31]:
```suggestion
In this **KafkaCluster** custom resource (https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster_with_ssl_hybrid_customcert.yaml) SSL is enabled for all listeners and certificates will be auto generated for "inner" and "controller" listeners. The "external" and "internal" listeners will use the user provided certificates. **ServerSSLCertSecret** is a reference to the Kubernetes secret that contains the server certificate for the listener to be used for SSL communication.
```

File: docs/ssl.md [hunk 3]
```diff
@@ -8,9 +8,11 @@ The {{< kafka-operator >}} makes securing your Apache Kafka cluster with SSL sim
 
 ## Enable SSL encryption in Apache Kafka {#enable-ssl}
 
-To create an Apache Kafka cluster with SSL encryption enabled, you must enable SSL encryption and configure the secrets in the **listenersConfig** section of your **KafkaCluster** Custom Resource. You can provide your own certificates, or instruct the operator to create them for you from your cluster configuration.
+To create an Apache Kafka cluster with SSL encryption enabled, you must enable SSL encryption and configure the secrets in the **listenersConfig** section of your **KafkaCluster** Custom Resource. You can provide your own CA cert and CA key, or instruct the operator to create them for you from your cluster configuration. Using **ssLSecrets** Koperator  generates client and server certificates signed by the provided or genetared CA. Listeners will share the same server certificate. The client certificate will be used by the Koperator, Cruise Control and Cruise Control Metrics Reporter to communicate with listener on SSL which is used for inner broker communication.
```
Comment (chrisgacsal) [inline docs/ssl.md:11]:
```suggestion
To create an Apache Kafka cluster which has listener(s) with SSL encryption enabled, you must enable SSL encryption and configure the secrets in the **listenersConfig** section of your **KafkaCluster** Custom Resource. You can either provide your own CA certificate and the corresponding private key or let the operator to create them for you from your cluster configuration. Using **sslSecrets** Koperator generates client and server certificates signed using the CA. The server certificate is shared across listeners. The client certificate is used by Koperator, Cruise Control and Cruise Control Metrics Reporter to communicate with Kafka brokers using listener with SSL enabled.
```

File: docs/ssl.md [hunk 4]
```diff
@@ -8,9 +8,11 @@ The {{< kafka-operator >}} makes securing your Apache Kafka cluster with SSL sim
...
+From Koperator 0.21.0+ you can also add your own certificates per listener. You can use hybrid solutions like you add your own certificate for one external listener and other listeners can use the auto generated one. See details below.
```
Comment (chrisgacsal) [inline docs/ssl.md:12]:
```suggestion
Providing custom certificates per listener is supported from Koperator version 0.21.0+. Having configurations where certain external listeners use user provided certificates while others rely on the auto-generated ones provided by Koperator are also supported. See details below.
```

**Comment by bartam1** [comment]
Thank you for the reivews!

Please use this PR for further reviews: https://github.com/banzaicloud/koperator-docs/pull/52.  It contains Robert work and my updates also

**Review by baluchicken — APPROVED** [review]

**Review by fekete-robert — APPROVED** [review]

**Pull request merged**